### PR TITLE
char mode for suite BabyAI

### PR DESCRIPTION
**Pull request by emailweixu**
The ppo_babyai.py example in #764 needs this.

### CODE REVIEW

File: alf/environments/suite_babyai.py [hunk 1]
```diff
@@ -186,15 +187,19 @@ def __init__(self,
...
         """Convert instruction string to a numpy array."""
+        if self._mode == 'char':
+            tokens = np.array([ord(c) for c in instruction])
+            return tokens
+
         tokens = self._word_pattern.findall(instruction.lower())
-        instr = np.array([self._vocab.get(token, 0) for token in tokens])
-        if np.amin(instr) == 0:
+        tokens = np.array([self._vocab.get(token, 0) for token in tokens])
+        if np.amin(tokens) == 0:
```
Comment (le-horizon) [inline alf/environments/suite_babyai.py]: This is an unsafe rename -- collides with the line below which also references "tokens" as the words before being translated into word_ids.
Let's name it token_ids or something like that?

**Review by le-horizon — COMMENTED** [review]
Looks almost good, just one thing to fix.

Thanks,
Le

File: alf/environments/suite_babyai.py [hunk 2]
```diff
@@ -172,8 +171,10 @@ def __init__(self,
             'mission':
                 gym.spaces.MultiDiscrete([vocab_size] * max_instruction_length)
         }
-        if one_token_per_step:
+        if mode == 'word':
             obs_space['mission'] = gym.spaces.Discrete(vocab_size)
+        elif mode == 'char':
+            obs_space['mission'] = gym.spaces.Discrete(128)
```
Comment (Haichao-Zhang) [inline alf/environments/suite_babyai.py:179]: Might be able to add a comment on using UTF-8 here or in the ```Arg:``` section?

File: alf/environments/suite_babyai.py (hunk 1 above)
Comment (emailweixu) [reply alf/environments/suite_babyai.py]: fixed.

File: alf/environments/suite_babyai.py (hunk 2 above)
Comment (emailweixu) [reply alf/environments/suite_babyai.py:179]: Added

**Review by Haichao-Zhang — APPROVED** [review]

**Pull request merged**